secant/Util/RecoveryPhraseStorage.swift

```diff
@@ -0,0 +1,238 @@
+    }
+
+    private static let zcashSeedPhrase = "zcashSeedPhrase"
+    private static let zcashBirthday = "zcashBirthday"
+    
+    // MARK: - Seed Phrase Public Interface
+    
+    func setSeedPhrase(_ phrase: String) throws {
+        guard let data = phrase.data(using: .utf8) else {
+            throw KeychainError.dataEncodingFailed
+        }
+            
+        try setData(data, forKey: RecoveryPhraseStorage.zcashSeedPhrase)
+    }
+
+    func seedPhrase() throws -> String {
+        guard let data = data(forKey: RecoveryPhraseStorage.zcashSeedPhrase) else {
+            throw KeychainError.noDataFound
+        }
+        
+        return String(decoding: data, as: UTF8.self)
+    }
+    
+    /// Use carefully:  Deletes seed phrase from the keychain!!!
+    @discardableResult
+    func deleteSeedPhrase() -> Bool {
+        return deleteData(forKey: RecoveryPhraseStorage.zcashSeedPhrase)
+    }
+
+    // MARK: - Birthday Public Interface
+
+    func setBirthday(_ birthday: BlockHeight) throws {
+        do {
+            let data = try NSKeyedArchiver.archivedData(withRootObject: NSNumber(value: birthday), requiringSecureCoding: true)
+            
+            try setData(data, forKey: RecoveryPhraseStorage.zcashBirthday)
+        } catch {
+            throw error
+        }
+    }
+
+    func birthday() throws -> BlockHeight {
+        guard let data = data(forKey: RecoveryPhraseStorage.zcashBirthday) else {
+            throw KeychainError.noDataFound
+        }
+
+        do {
+            if let birthday = try NSKeyedUnarchiver.unarchivedObject(ofClass: NSNumber.self, from: data) {
+                return birthday.intValue
+            }
+            
+            throw KeychainError.dataDecodingFailed
+        } catch {
+            throw error
+        }
+    }
+    
+    /// Use carefully:  Deletes birthday from the keychain!!!
+    @discardableResult
+    func deleteBirthday() -> Bool {
+        return deleteData(forKey: RecoveryPhraseStorage.zcashBirthday)
+    }
+    
+    // MARK: - Private
+
+    /// Store data for key
+    private func setData(
+        account: String = "",
+        _ data: Data,
+        forKey: String
+    ) throws {
+        let query:[ String: AnyObject ] = [
+            /// Data to be stored
+            kSecValueData as String: data as AnyObject,
+            /// Uniquely identify this keychain accessor
+            kSecAttrService as String: forKey as AnyObject,
+            kSecAttrAccount as String: account as AnyObject,
+            kSecClass as String: kSecClassGenericPassword,
+            /// The data in the keychain item can be accessed only while the device is unlocked by the user.
+            /// This is recommended for items that need to be accessible only while the application is in the foreground.
+            /// Items with this attribute do not migrate to a new device.
+            /// Thus, after restoring from a backup of a different device, these items will not be present.
```

Comment:
@geffenz @defuse  This is relevant in terms of backup and restore experience. If this is the accessibility we want for this keychain item (I'm pretty sure it is), we should communicate the side-effects to the users.

Reply:
FYI:
There are several options of accessibility and for now I used the most protected and secure one possible. Other potential option is to set the accessibility to `whenUnlocked`. The result would be:

`This is recommended for items that need to be accessible only while the application is in the foreground. Items with this attribute migrate to a new device when using encrypted backups.`

I think these two options are the only possible for our case so the discussion should be based on "Do we want to allow migration of the stored seed and birthday from device to device and rely on Apple's encryption?". I personally experienced both, one wallet was seamlessly restored when I restored from iCloud backup while other requested to import the seed and start over.